src/tooltip.tsx

```diff
@@ -112,386 +161,442 @@ const Tooltip: React.FC<Props> = ({
     description,
     target,
     title,
-    targetLabel,
-    delay = true,
+    width,
+    position = 'top',
     dataAttributes,
-    targetStyle,
-    unstable_offsetX,
+    delay = true,
+    centerContent,
     textCenter,
-    ...rest
 }) => {
-    const {isDarkMode} = useTheme();
-    const [isVisible, setIsVisible] = React.useState(false);
-    const {isTabletOrSmaller} = useScreenSize();
-    const ariaId = useAriaId();
-    const isPointerOver = React.useRef(false);
-    const closeTooltipTimeoutId = React.useRef<NodeJS.Timeout | null>(null);
-    const showTooltipTimeoutId = React.useRef<NodeJS.Timeout | null>(null);
-    const targetRef = React.useRef<HTMLDivElement>(null);
-    const tooltipRef = React.useRef<HTMLDivElement>(null);
-    const targetBoundingClientRect = React.useRef({
-        top: 0,
-        right: 0,
-        left: 0,
-        bottom: 0,
-        width: 0,
-        height: 0,
-    });
+    const tooltipId = useAriaId();
+    const {openTooltipId} = useTooltipState();
+    const {open, close} = useSetTooltipState();
 
-    const [containerPosition, setContainerPosition] = React.useState({});
+    const [tooltipComputedProps, setTooltipComputedProps] = React.useState<{
+        left: number;
+        top: number;
+        padding: string;
+    } | null>(null);
 
-    const getPosition = (position: Position | undefined) =>
-        isTabletOrSmaller ? position || defaultPositionMobile : position || defaultPositionDesktop;
+    const [arrowComputedProps, setArrowComputedProps] = React.useState<React.CSSProperties>({});
 
-    const [tooltipClientRect, setTooltipClientRect] = React.useState<DOMRect | undefined>(undefined);
+    const targetRef = React.useRef<Element | null>(null);
+    const tooltipRef = React.useRef<HTMLDivElement | null>(null);
+    const [tooltip, setTooltip] = React.useState<HTMLElement | null>(null);
+    const isTouchableDevice = isClientSide() ? window.matchMedia('(pointer: coarse)').matches : false;
 
-    const validatePosition = (position: Position) => {
-        if (!tooltipClientRect) return position;
+    const [isMouseOverTooltip, setIsMouseOverTooltip] = React.useState(false);
+    const [isMouseOverTarget, setIsMouseOverTarget] = React.useState(false);
+    const [isFocused, setIsFocused] = React.useState(false);
+    const [isTooltipOpen, setIsTooltipOpen] = React.useState(false);
+    const isInverse = useIsInverseVariant();
 
-        const validatePositionLeft = (position: Position) => {
-            const hasTopSpace = targetBoundingClientRect.current.top > tooltipClientRect.height;
-            return targetBoundingClientRect.current.left < tooltipClientRect.width
-                ? hasTopSpace
-                    ? 'top'
-                    : 'bottom'
-                : position;
-        };
+    const targetRect = useBoundingRect(targetRef);
```

Comment:
If I understand this, you're tracking the client rect with an infinite requestAnimationFrame loop even when the tooltip is not open. Ideally we should only have the rAF loop active while the tooltip is open

Reply:
updated useBoundingRect to optionally avoid tracking the bounding rectangle on every frame